src/python/pants/backend/python/rules/python_test_runner_integration_test.py

```diff
@@ -1,266 +1,376 @@
-
-
-        testprojects/tests/python/pants/dummies:target_with_thirdparty_dep              .....   SUCCESS
-        """)
     )
-
-  def test_transitive_dep(self):
-    pants_run = self.run_passing_pants_test([
-      'testprojects/tests/python/pants/dummies:target_with_transitive_dep',
-    ])
-    self.assert_fuzzy_string_match(
-      pants_run.stdout_data,
-      dedent("""\
-        testprojects/tests/python/pants/dummies:target_with_transitive_dep stdout:
-        ============================= test session starts ==============================
-        platform SOME_TEXT
-        rootdir: SOME_TEXT
-        plugins: SOME_TEXT
-        collected 1 item
-
-        pants/dummies/test_with_transitive_dep.py .                              [100%]
-
-        ============================== 1 passed in SOME_TEXT ===============================
-
-
-        testprojects/tests/python/pants/dummies:target_with_transitive_dep              .....   SUCCESS
-        """)
+    source = FileContent(
+      path="test_transitive_dep.py",
+      content=dedent(
+        """\
+        from pants_test.transitive_dep import add_four
+
+        def test():
+          assert add_four(2) == 6
+        """
+      ).encode(),
     )
-
-  def test_respects_passthrough_args(self):
-    pants_run = self.run_passing_pants_test([
-      '--pytest-args=-k test_run_me',
-      'testprojects/tests/python/pants/dummies:needs_config',
-    ])
-    self.assert_fuzzy_string_match(
-      pants_run.stdout_data,
-      dedent("""\
-        testprojects/tests/python/pants/dummies:needs_config stdout:
-        ============================= test session starts ==============================
-        platform SOME_TEXT
-        rootdir: SOME_TEXT
-        plugins: SOME_TEXT
-        collected 2 items / 1 deselected / 1 selected
-
-        pants/dummies/test_config_works.py .                                     [100%]
-
-        ======================= 1 passed, 1 deselected in SOME_TEXT ========================
-
-
-        testprojects/tests/python/pants/dummies:needs_config                            .....   SUCCESS
-        """)
+    self.create_python_test_target([source], dependencies=[":transitive_dep"])
+    result = self.run_pytest()
+    assert result.status == Status.SUCCESS
+    assert "test_transitive_dep.py ." in result.stdout
+
+  def test_thirdparty_dep(self) -> None:
+    self.setup_thirdparty_dep()
+    source = FileContent(
+      path="test_3rdparty_dep.py",
+      content=dedent(
+        """\
+        from ordered_set import OrderedSet
+
+        def test():
+          assert OrderedSet((1, 2)) == OrderedSet([1, 2])
+        """
+      ).encode(),
+    )
+    self.create_python_test_target([source], dependencies=["3rdparty/python:ordered-set"])
+    result = self.run_pytest()
+    assert result.status == Status.SUCCESS
+    assert "test_3rdparty_dep.py ." in result.stdout
+
+  def test_thirdparty_transitive_dep(self) -> None:
```

Comment:
This is a new test. I added it because it originally took some work to get the Python test runner to resolve transitive 3rd party deps. This will ensure we never lose that support.